zerver/models.py

```diff
@@ -1896,6 +1904,14 @@ class UserProfile(AbstractBaseUser, PermissionsMixin, UserBaseSettings):  # type
 
     objects = UserManager()
 
+    # Contains a external uid of the user to make it able
+    # to sync mails form e.g. LDAP
+    ext_auth_id: Optional[ExtAuthId] = models.OneToOneField(
+        ExtAuthId, on_delete=models.CASCADE, null=True
+    )
+
+    objects: UserManager = UserManager()
```

Comment:
I'm tempted to say we should just not have this field and make the external auth IDs be their own independent table, rather than adding a `OneToOneField` on `UserProfile`. I'll post notes on how that might work.

(BTW this line is a rebase mistake copying one a few lines up)